wrapper/src/main/java/com/moodysalem/phantomjs/wrapper/PhantomJS.java

```diff
@@ -88,15 +69,43 @@ public static InputStream render(final InputStream html, final RenderOptions opt
      *             if any file operations fail
      * @throws RenderException
      *             if the render script fails for any reason
+     * @deprecated This method will be deleted with the next version. Please use {@link PhantomJS#render(InputStream, RenderOptions)} instead.
      */
-    public static InputStream render(final PhantomJSOptions options, final InputStream html, final PaperSize paperSize, final ViewportDimensions dimensions, final Margin margin, final BannerInfo headerInfo, final BannerInfo footerInfo,
-            final RenderFormat renderFormat, final Long jsWait, final Long jsInterval) throws IOException, RenderException {
-        if (html == null || renderFormat == null || paperSize == null || dimensions == null || margin == null || headerInfo == null || footerInfo == null || jsWait == null || jsInterval == null) {
+    @Deprecated
+    public static InputStream render(final PhantomJsOptions options, final InputStream html, final PaperSize paperSize, final ViewportDimensions dimensions, final Margin margin, final BannerInfo headerInfo, final BannerInfo footerInfo,
+            final RenderFormat renderFormat, final Long jsWait, final Long jsInterval)
+        throws IOException, RenderException, RenderOptionsException {
+
+        RenderOptions renderOptions = new RenderOptions().phantomJSOptions(options).paperSize(paperSize).margin(margin).viewportDimensions(dimensions).headerInfo(headerInfo).footerInfo(footerInfo).renderFormat(renderFormat).jsWait(jsWait).jsInterval(jsInterval);
+
+        return render(html, renderOptions);
+    }
+
+
+    /**
+     * Another way to call PhantomJS#render using the RenderOptions to specify
+     * all the common options
+     *
+     * @param html
+     *            to render
+     * @param options
+     *            for rendering
+     * @return a stream of the rendered output
+     * @throws IOException
+     *             if anything goes wrong executing the program
+     * @throws RenderException
+     *             if the render script fails for any reason
+     */
+    public static InputStream render(final InputStream html, final RenderOptions options)
+        throws IOException, RenderException, RenderOptionsException {
+        if (html == null || options == null) {
             throw new NullPointerException("All parameters are required");
         }
 
-        if (jsWait < 0 || jsInterval < 0 || (jsWait > 0 && jsInterval > jsWait) || (jsInterval == 0 && jsWait > 0)) {
-            throw new IllegalArgumentException("Invalid jsWait or jsInterval values provided");
+        if(!options.isValid()) {
```

Comment:
I see your point. However I like the idea of the beans holding their own constraints instead of keeping the constraints elsewhere in the code. Especially stuff like the constraint between jsExecutionTimeout and jsInterval.